improver/threshold.py

```diff
@@ -246,7 +247,7 @@ def _add_threshold_coord(self, cube, threshold):
                 With new "threshold" axis
         """
         coord = iris.coords.DimCoord(
-            np.array([threshold], dtype=np.float32), units=cube.units
+            np.array([threshold], dtype=FLOAT_DTYPE), units=cube.units
```

Comment:
Good point.  I updated it here because there's now a central definition of the datatype we expect.  Most of the other files I've updated in these PRs are unit tests, which need to fit with what's required by the `set_up_test_cubes` module - which at the moment is hard-coded as float32.  But in the longer term we should probably tidy this up and use `FLOAT_DTYPE` everywhere.